src/fsharp/IlxGen.fs

```diff
@@ -2198,12 +2198,12 @@ and GenCoerce cenv cgbuf eenv (e,tgty,m,srcty) sequel =
      // Do an extra check - should not be needed 
      TypeRelations.TypeFeasiblySubsumesType 0 cenv.g cenv.amap m tgty TypeRelations.NoCoerce srcty then
      begin 
-       // The .NET IL doesn't always support implict subsumption for interface types, e.g. at stack merge points 
-       // Hence be conservative here and always cast explicitly. 
        if (isInterfaceTy cenv.g tgty) then (
            GenExpr cenv cgbuf eenv SPSuppress e Continue
            let ilToTy = GenType cenv.amap m eenv.tyenv tgty
-           CG.EmitInstrs cgbuf (pop 1) (Push [ilToTy]) [ I_unbox_any ilToTy  ]
+           // I believe section "III.1.8.1.3 Merging stack states" of ECMA-335 implies that no unboxing
```

Comment:
haha, well I was hoping someone who confirm that, and then it wouldn't be total based on my potentially misaken belief!